### ISSUE TITLE
Bug in Bignum#bit_length, fails with TypeError: nil can't be coerced into Fixnum for bit lengths divisible by 256

### ISSUE THREAD

**Issue by grempe** [marcandre/backports#95]
In Ruby 2.0.0 (and others prior to 2.1 where #bit_length is not natively supported) calling bit_length on a Bignum using the backported version fails every time when called on a number that is one of the following exact bit lengths. If you add or subtract one bit of length in the Bignum it works fine.

FAILS:
256, 512, 768, 1024, 2048

At first I thought this was failing on powers of 2.  But it appears to also fail on any multiple of 256 (e.g. 768).

Running on numbers even one bit larger or smaller than the above works cleanly every time, with thousands of generated numbers tested.

Here is the function I am using to get exact bitlength random numbers:
https://github.com/grempe/secretsharing/blob/master/lib/secretsharing/shamir.rb#L28

The native version of Bignum#bit_length in Ruby 2.1+ does not have this problem. Only the backport.

And here is an example of the failing output and backtrace:

````
irb(main):288:0> 1000.times { r = get_random_number_with_bitlength(256); puts r; r.bit_length }
91178362617816881166579720176198217549251305244541026425489888079462471837807
TypeError: nil can't be coerced into Fixnum
	from /usr/local/var/rbenv/versions/2.0.0-p645/lib/ruby/gems/2.0.0/gems/backports-3.6.4/lib/backports/2.1.0/bignum/bit_length.rb:12:in `+'
	from /usr/local/var/rbenv/versions/2.0.0-p645/lib/ruby/gems/2.0.0/gems/backports-3.6.4/lib/backports/2.1.0/bignum/bit_length.rb:12:in `bit_length'
	from (irb):288:in `block in irb_binding'
	from (irb):288:in `times'
	from (irb):288
	from /usr/local/var/rbenv/versions/2.0.0-p645/bin/irb:12:in `<main>'
````

**Comment by grempe** [comment]
I found that this issue comes about as a results of the Range#bsearch backports code returning nil (its not 'n' that is nil when the error is thrown).

I created a simple monkey patch in my own code that seems to be doing the trick (temporarily?) instead of using backports version.

````
unless Bignum.method_defined? :bit_length
  class Bignum
    def bit_length
      (self).abs.to_s(2).length
    end
  end
end
````

**Comment by marcandre** [comment]
Fixed now in 3.6.5, thanks for reporting this :smile:
